### ISSUE TITLE
Upgrade JNA used for SBT Zinc

### ISSUE THREAD

**Issue by msogrin** [gradle/gradle#20361]
Gradle build on Macosx Monterey fails on `compileScala` task when executed from Intellij IDEA with error
```
Unable to load library 'c': Native library (darwin/libc.dylib) not found in resource path
```

The error doesn't happen when build is started from Terminal.
The error happens only for some team members, and other people with seemingly identical setup have no issues.

The error is similar to https://youtrack.jetbrains.com/issue/IDEA-244962. And the workaround suggested there helped: upgrade JNA to at least 5.6.

I upgraded to 5.11.0 using
```
dependencies {
    constraints {
        zinc("net.java.dev.jna:jna:5.11.0")
        zinc("net.java.dev.jna:jna-platform:5.11.0")
    }
}
```
and it helped to eliminate the error when executed from IDEA. No ill effects were observed yet but the testing wasn't extensive.

It would be better if Gradle distribution used a newer version of JNA by default without need for workaround in user build scripts.

**Comment by hegyibalint** [comment]
What happens if you would upgrade the `zinc` version?

Related documentation: https://docs.gradle.org/current/userguide/scala_plugin.html#sec:configure_zinc_compiler
Related issue: #19864

**Comment by msogrin** [comment]
I'll test when #19864 gets into a RC/released version.

**Comment by msogrin** [comment]
Tested with 7.5-rc1, the issue is resolved: Zinc 1.6.1 uses JNA 5.8.0, which is higher than required 5.6.
